core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -110,17 +110,28 @@ oppia.directive('searchBar', [function() {
         };
 
         var _searchBarFullyLoaded = false;
+        var url;
 
         var _hasChangedSearchQuery = Boolean(urlService.getUrlParams().q);
-        var _onSearchQueryChangeExec = function() {
-          searchService.executeSearchQuery(
+        var _onSearchQueryChangeExec = function(reloadPage) {
+          $scope.searchIsLoading = true;
+          url = searchService.executeSearchQuery(
               $scope.searchQuery, $scope.selectionDetails.categories.selections,
               $scope.selectionDetails.languageCodes.selections, function() {
             if (!_hasChangedSearchQuery && _searchBarFullyLoaded) {
               _hasChangedSearchQuery = true;
               $rootScope.$broadcast('hasChangedSearchQuery');
             }
           });
+          if ($window.location.search != url && reloadPage) {
+            $window.location.href = '/search\?q=' + url;
```

Comment:
It used to fail for some reason (my guess is because of setting `html5Mode`), it works for me now, so I'll remove the backslash.